### PR TITLE
Increase @safety of std.conv

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -1153,14 +1153,14 @@ if (!(isImplicitlyConvertible!(S, T) &&
...
 // https://issues.dlang.org/show_bug.cgi?id=16108
-@system unittest
+@safe unittest
 {
     static struct A
     {
         int val;
         bool flag;
 
-        string toString() { return text(val, ":", flag); }
```
Comment (dkorpel) [inline std/conv.d:1163]: Is `@safe` not inferred here?

**Comment by atilaneves** [comment]
@dkorpel - this makes it so that std.conv tests build with the changes from [your PR](https://github.com/dlang/dmd/pull/12989).

File: std/conv.d [hunk 2]
```diff
@@ -3122,6 +3122,12 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
         alias p = source;
     }
 
+    void advanceSource() @trusted {
```
Comment (dkorpel) [inline std/conv.d:3125]: `{` on new line

File: std/conv.d [hunk 3]
```diff
@@ -3168,8 +3175,7 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
         // skip past the last 'f'
         ++count;
         p.popFront();
-        static if (isNarrowString!Source)
-            source = cast(Source) p;
+        advanceSource;
```
Comment (dkorpel) [inline std/conv.d:3178]: Add `()` to function call

**Review by dkorpel — COMMENTED** [review]
Thanks!

File: std/conv.d (hunk 3 above)
Comment (atilaneves) [reply std/conv.d:3178]: Is that part of the style guide?

**Review by dkorpel — APPROVED** [review]

File: std/conv.d (hunk 3 above)
Comment (dkorpel) [reply std/conv.d]: It's not in [The D Style](https://dlang.org/dstyle.html), but I think de facto the style is to only omit braces for functional / property-like functions, and not omit them in 'regular' calls with side-effects.

**Pull request merged**